disabledModules not working when using HM as a NixOS module

### Issue description

I am trying to backport some modules from unstable in HM `release-20.09` in my WIP Flake based NixOS/HM configuration. I am trying to keep both HM standalone and as a NixOS module as close as possible in configuration.

The issue is in what I am trying to do here: https://github.com/thiagokokada/nix-configs/commit/e628d2ef21a77a8f00538a4a7aae6bc655102683#diff-3802086929d8fa5ef96d3fde6ef3deaa276509760bc03919b07552b960fcb076R5-R6. This works in standalone mode but not in HM as a NixOS module, returning this error:

```
$ nixos-rebuild build-vm --flake '.#desktop'
building the system configuration...
error: --- ThrownError -------------------------------------------------------------------------------------------------------------- nix
The option `home-manager.users.thiagoko.programs.mpv.enable' in `/nix/store/v9ngxqny70y7chhc9brp3pqwx2jcns4n-source/modules/programs/mpv.nix' is already declared in `/nix/store/l82yrf9fpmpzhy125a4y1g8yqsj6a7i3-source/modules/programs/mpv.nix'.
(use '--show-trace' to show detailed location information)
make: *** [Makefile:9: build-vm-desktop] Error 1
```

I also tried to set `disabledModules` as:

```
disabledModules = [ "${inputs.home}/modules/programs/mpv.nix" ];
```

I even tried:

```
disabledModules = [ "/nix/store/l82yrf9fpmpzhy125a4y1g8yqsj6a7i3-source/modules/programs/mpv.nix" ];
```

However I got the same result.

### Meta

#### Maintainer CC

@rycee @berbiche 

#### Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 5.8.7-zen1, NixOS, 20.09.3072.d4c29df154d (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20201201_5a6ddb3`
 - channels(root): `"nixos-20.09.3072.d4c29df154d"`
 - channels(thiagoko): `"home-manager-20.09, nixos-20.09.2947.2b9daa020d4, nixos-unstable, nixpkgs-unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

--- a/nixos/default.nix
+++ b/nixos/default.nix
@@ -12,6 +12,7 @@ let
     specialArgs = {
       lib = extendedLib;
       nixosConfig = config;
+      modulesPath = ../modules;
     } // cfg.extraSpecialArgs;
     modules = [
       ({ name, ... }: {
@@ -72,7 +73,6 @@ in {
       extraSpecialArgs = mkOption {
         type = types.attrs;
         default = { };
-        example = literalExample "{ modulesPath = ../modules; }";
         description = ''
           Extra <literal>specialArgs</literal> passed to Home Manager.
         '';

